frontend/src/types/vuex.d.ts

```diff
@@ -0,0 +1,64 @@
+import { Store } from 'vuex';
+import { AuthorshipFile, User } from './types';
+
+interface AuthorshipInfo {
+  author: string;
+  checkedFileTypes?: string[];
+  files: AuthorshipFile[];
+  isMergeGroup: boolean;
+  isRefresh?: boolean;
+  location: string | undefined;
+  maxDate: string;
+  minDate: string;
+  name: string;
+  repo: string;
+}
+
+interface ZoomInfo {
+  isRefreshing?: boolean;
+  zAuthor: string;
+  zAvgCommitSize: number | string;
+  zFileTypeColors: { [key: string]: string };
+  zFilterGroup: string;
+  zFilterSearch: string;
+  zFromRamp: boolean;
+  zIsMerged: boolean;
+  zLocation: string | undefined;
+  zRepo: string;
+  zSince: string;
+  zTimeFrame: string;
+  zUntil: string;
+  zUser: User;
+}
+
+interface SummaryDates {
+  since: string;
+  until: string;
+}
+
+interface StoreState {
+  tabAuthorshipInfo: AuthorshipInfo;
+  tabZoomInfo: ZoomInfo;
+  summaryDates: SummaryDates;
+  mergedGroups: string[]
+  fileTypeColors: { [key: string]: string }
+  loadingOverlayCount: number
+  loadingOverlayMessage: string
+  isTabActive: boolean
+}
+
+declare module '@vue/runtime-core' {
+  interface State {
```

Comment:
Would it be possible to reuse the `StoreState` interface here?

Reply:
Pushed a commit for the state to extend the `StoreState` interface, thanks for pointing this out!